wptdash/templates/performance.html

```diff
@@ -0,0 +1,175 @@
+              <svg id="cumulative-chart" width="500", height="250"></svg>
+            </td>
+          </tr>
+          <tr>
+            <td>15 minutes</td>
+            <td>{{ jobs_by_delta[15]|int }}%</td>
+          </tr>
+          <tr>
+            <td>30 minutes</td>
+            <td>{{ jobs_by_delta[30]|int }}%</td>
+          </tr>
+          <tr>
+            <td>1 hour</td>
+            <td>{{ jobs_by_delta[60]|int }}%</td>
+          </tr>
+          <tr>
+            <td>2 hours</td>
+            <td>{{ jobs_by_delta[120]|int }}%</td>
+          </tr>
+          <tr>
+            <td>4 hours</td>
+            <td>{{ jobs_by_delta[240]|int }}%</td>
+          </tr>
+          <tr>
+            <td>24 hours</td>
+            <td>{{ jobs_by_delta[1440]|int }}%</td>
+          </tr>
+        </tbody>
+      </table>
+      <h2>Distribution of Wait and Build Times</h2>
+      <svg id="wait-times" height="250" width="400"></svg>
+      <svg id="build-times" height="250" width="400"></svg>
+      <h2>PRs taking over 1 hour</h2>
+      <table class="table">
+        <thead>
+          <tr>
+            <th>Pull Request</th>
+            <th>Title</th>
+            <th>Total Time</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for pr in outlier_prs %}
+            <tr>
+              <td>{{ pr['number'] }}</td>
+              <td><a href="https://github.com/w3c/web-platform-tests/pull/{{pr['number']}}">{{ pr['title'] }}</a></td>
+              <td>{{ pr['time'] }}</td>
+            </tr>
+          {% endfor %}
+        </tbody>
+      </table>
+      <h2>All First-Completed Jobs for each PR</h2>
+      <table class="table">
+        <thead>
+          <tr>
+            <th>Pull Request</th>
+            <th>Job</th>
+            <th>Build Wait Time</th>
+            <th>Job Duration</th>
+            <th>Total Time</th>
+            <th>Job Status</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for job in jobs %}
+            {% set job_finished_at = job.finished_at or job.build.finished_at or now %}
+            {% set build_wait = (job.build.started_at or job.build.pull_request.created_at) - job.build.pull_request.created_at %}
+            {% set total_time = job_finished_at - job.build.pull_request.created_at %}
+            {% set job_delta = total_time - build_wait %}
+            <tr>
+              <td>{{ job.build.pull_request.number }}</td>
+              <td>{{ job.product.name|replace(':', ' ')|title }}</td>
+              <td>{{ build_wait }}</td>
+              <td>{{ job_delta }}</td>
+              <td>{{ total_time }}</td>
+              <td>{{ job.state.name }}</td>
+            </tr>
+          {% endfor %}
+        </tbody>
+      </table>
+    {% endif %}
+    {% autoescape false %}
+    <script>
+      window.WPTDASH = {
+        cumulativeData: '{{ cumulative_chart_data }}',
+        buildTimes: '{{ build_times }}',
```

Comment:
This is a bit curious. It gets parsed as JSON, so why not just include it unquoted and let the JS parser do it? Saves a bit string in memory, and less code.

Reply:
:+1: As soon as I start using `json.dumps` I think in strings, even when I don't have to.